yt_dlp/update.py

```diff
@@ -146,179 +164,290 @@ def _sha256_file(path):
+            result_version = None
+
+        checksum = None
+        # Non-updateable variants can get update_info but need to skip checksum
+        if not is_non_updateable():
+            try:
+                hashes = self._download_asset('SHA2-256SUMS', result_tag)
+            except network_exceptions as error:
+                if not isinstance(error, HTTPError) or error.status != 404:
+                    self._report_network_error(f'fetch checksums: {error}')
+                    return None
+                self.ydl.report_warning('No hash information found for the release, skipping verification')
+            else:
+                for ln in hashes.decode().splitlines():
+                    if ln.endswith(_get_binary_name()):
+                        checksum = ln.split()[0]
+                        break
+                if not checksum:
+                    self.ydl.report_warning('The hash could not be found in the checksum file, skipping verification')
+
+        if _output:
+            requested_label = _make_label(self.requested_repo, resolved_tag, requested_version)
+            update_label = _make_label(self.requested_repo, result_tag, result_version)
+            self.ydl.to_screen(
+                f'Current version: {current_label}\n'
+                + f'{"Latest" if self.requested_tag == "latest" else "Requested"} version: {requested_label}'
+                + (f'\nUpgradable to: {update_label}' if update_label != requested_label else ''))
+
+        return UpdateInfo(
+            tag=result_tag,
+            version=result_version,
+            requested_version=requested_version,
+            commit=target_commitish if result_tag == resolved_tag else None,
+            checksum=checksum)
+
+    def update(self, update_info=NO_DEFAULT):
         """Update yt-dlp executable to the latest version"""
-        if not self.check_update():
-            return
+        if update_info is NO_DEFAULT:
+            update_info = self.query_update(_output=True)
+        if not update_info:
+            return False
+
         err = is_non_updateable()
         if err:
-            return self._report_error(err, True)
-        self.ydl.to_screen(f'Updating to {self._label(self.target_channel, self.new_version)} ...')
-        if (_VERSION_RE.fullmatch(self.target_tag[5:])
-                and version_tuple(self.target_tag[5:]) < (2023, 3, 2)):
-            self.ydl.report_warning('You are downgrading to a version without --update-to')
-            self._block_restart('Cannot automatically restart to a version without --update-to')
+            self._report_error(err, True)
+            return False
+
+        self.ydl.to_screen(f'Current Build Hash: {_sha256_file(self.filename)}')
+        if not update_info.checksum:
+            self._block_restart('Automatically restarting into unverified builds is disabled for security reasons')
+
+        update_label = _make_label(self.requested_repo, update_info.tag, update_info.version)
```

Comment:
```suggestion

        update_label = _make_label(self.requested_repo, update_info.tag, update_info.version)
```

Reply:
Why?

Reply:
the removed lines were moved further down